#ifndef MAINWINDOW_H
#define MAINWINDOW_H

#include "configmanager.h"
#include <QMainWindow>

QT_BEGIN_NAMESPACE
namespace Ui {
class MainWindow;
}
QT_END_NAMESPACE

class MainWindow : public QMainWindow
{
    Q_OBJECT

public:
    MainWindow(QWidget *parent = nullptr);
    ~MainWindow();

private slots:
    void on_actionNew_triggered();
    void on_actionOpen_triggered();

private:
    Ui::MainWindow *ui;
    ConfigManager config;
    void print(const QString &msg, const QString &type = "info");
    void openFile(const QString &filePath);
    void openNewFile();
};
#endif // MAINWINDOW_H
